arch-gulpfile.js

```diff
@@ -130,6 +133,47 @@ function generateServiceWorker() {
     return false;
   }
 }
+
+function inlineCriticalCSS(cb) {
+  const HOST = process.env.HOST || 'localhost';
+  const PORT = process.env.PORT || 3000;
+  const PATH = process.env.CRITICAL_PATH || '/';
+  const url = `http://${HOST}:${PORT}${PATH}`;
+  const PARSE_TIMEOUT = 2000 // How long to wait for the server to start
+  const statsPath = Path.resolve(process.cwd(), 'dist/server/stats.json');
+  const stats = JSON.parse(fs.readFileSync(statsPath));
+  const cssAsset = stats.assets.find(({ name }) => name.endsWith('.css'));
+  const cssAssetPath = Path.resolve(process.cwd(), `dist/js/${cssAsset.name}`);
+  const targetPath = Path.resolve(process.cwd(), 'dist/js/critical.css');
+  const penthouseOptions = {
+    url: url,
+    css: cssAssetPath,
+    width: 1440,
+    height: 900,
+    timeout: 30000,
+    strict: false,
+    maxEmbeddedBase64Length: 1000,
+    renderWaitTime: 2000,
+    blockJSRequests: false,
+  };
+  const serverProcess = spawn('node', ['server/index.js']);
+  // Use setTimeout so the server has enough time to bind to the port
+  setTimeout(() => {
+    penthouse(penthouseOptions, (err, css)  => {
+        if (err) {
+          throw err;
+        }
+        const minifiedCSS = new CleanCSS().minify(css).styles;
+        fs.writeFile(targetPath, minifiedCSS, (err) => {
+          if (err) {
+            throw err;
+          }
+          serverProcess.kill(0);
+          cb();
+        })
+    });
+  }, PARSE_TIMEOUT);
```

Comment:
another option is to export the server startup promise in `server/index.js` https://github.com/electrode-io/electrode-boilerplate-universal-react-node/blob/master/server/index.js#L58

and just directly `require("./server/index.js");` instead of spawn it.